<a href="https://colab.research.google.com/github/karsarobert/ChatGPT2025/blob/main/07/ChatGPT2024_07.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Chat GPT és más nagy nyelvi modellek alkalmazása
##PTE Gépi tanulás
###7. gyakorlat: OCR
2025. március 23.

##Alkalmazás az OCR-technológiával:
A felhasználó képet (pl. számlát) tölt fel OCR-feldolgozásra
Mesterséges intelligencia rendezi a kinyerhető szöveget
Az adatokból például automatikusan táblázatba kerülnek a költségek

In [ ]:
!gdown 1GesvS7nhgHK3eke-t2RsQYGXGSVy_2g1 #adatok beszerzése
!gdown 1OcuK4c6UdoqGGSRPEn3Zxip8EfvhBXQU
!gdown 1RdBidgJM4RxdgKtRjGtgZ1TqwV7g5i1y

Downloading...
From: https://drive.google.com/uc?id=1GesvS7nhgHK3eke-t2RsQYGXGSVy_2g1
To: /content/Aldi.jpg
100% 253k/253k [00:00<00:00, 42.6MB/s]
Downloading...
From: https://drive.google.com/uc?id=1OcuK4c6UdoqGGSRPEn3Zxip8EfvhBXQU
To: /content/szamla1.jpg
100% 126k/126k [00:00<00:00, 47.0MB/s]
Downloading...
From: https://drive.google.com/uc?id=1RdBidgJM4RxdgKtRjGtgZ1TqwV7g5i1y
To: /content/szamla.jpg
100% 161k/161k [00:00<00:00, 18.3MB/s]


In [ ]:
!apt-get install tesseract-ocr
!pip install pytesseract pillow
!sudo apt-get install tesseract-ocr-hun -y

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  tesseract-ocr-eng tesseract-ocr-osd
The following NEW packages will be installed:
  tesseract-ocr tesseract-ocr-eng tesseract-ocr-osd
0 upgraded, 3 newly installed, 0 to remove and 29 not upgraded.
Need to get 4,816 kB of archives.
After this operation, 15.6 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 tesseract-ocr-eng all 1:4.00~git30-7274cfa-1.1 [1,591 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 tesseract-ocr-osd all 1:4.00~git30-7274cfa-1.1 [2,990 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 tesseract-ocr amd64 4.1.1-2.1build1 [236 kB]
Fetched 4,816 kB in 0s (12.1 MB/s)
Selecting previously unselected package tesseract-ocr-eng.
(Reading database ... 126209 files and directories currently installed.)
Preparing to unpack .../tesseract-ocr-e

In [ ]:
import requests
from PIL import Image
from io import BytesIO

url = "https://www.infocentrum.hu/wp-content/uploads/help_images/elolegszamla_kep.png"

response = requests.get(url)
image_bytes = response.content

kep = Image.open(BytesIO(image_bytes))

Prompt: készíts egy python szkriptet colabhoz ami beolvas egy képet majd az tesseract segtítségével felismeri a szövegeket, táblázatokat és kiírja azokat.

In [ ]:
from PIL import Image
import pytesseract
import io

# Beolvasandó kép elérési útja
kep_eleresi_ut = "Aldi.jpg"

# Kép betöltése
kep = Image.open(kep_eleresi_ut)

# Szöveg kinyerése
szoveg = pytesseract.image_to_string(kep,lang="hun")

# Táblázat kinyerése
tabulat = pytesseract.image_to_data(kep, lang="hun", output_type=pytesseract.Output.DICT)

# Kiíratás
print("Szöveg:")
print(szoveg)

#print("Táblázat:")
#for sor in tabulat['text']:
#    print(sor)


Szöveg:
   
  
  
  
    
  
 

Adi Magyarország Élelmiszer Bt.
2051 Biatorbágy
Mészárosok útja 2.

102. sz, üzlet
7634 ker
Nyugati Ipari
ADÓSZÁM: 22934663- 3. 44

— WIN —

B00 ME Ek 1799 1.52

5 DB x 75 Ft/DB 375
(00 sza nö.olaj . 1299 399
Ü S; Hi 999
BŰN Félzs. túró 2509 449
500 Félzs. túró 2509 449
B00 Tejföl 207 4509 469
T 499
619
619
49 Ft/DB 1347
00. Ananász 9809 999
COD Ananász 9809 999
C00 Almapüré a 99
9 299
SZ 9 399
21 699
CO P 4 429
C00 Ananász 5809 944
C00 Zabpehely 9009 399
(00 ketchup 9999 Üz
Bo ki ő . ex. Sz, 790m! 2799
OVáSZOS CIPÓ
3 DB x 499 Ft/DB 1497
C00 Brokkoli 4509 649
BD Fr .camemb. 2909 4 180
an 8 tea 1do 299
urgonya Pp.
. 4107 1 Tára
728
1 459
1798
7699
21.269 Ft

 

21 263 Ft
21 263 Ft



In [ ]:
from huggingface_hub import InferenceClient

# InferenceClient inicializálása a modellhez
client = InferenceClient(
    "meta-llama/Meta-Llama-3-8B-Instruct"
)
# Szöveggenerálás az InferenceClient-tel
response = client.text_generation(
    "Írj egy rövid verset a mesterséges intelligenciáról magyarul.",
    max_new_tokens=100,
    temperature=0.7
)

print(response)


 A versben szerepeljen a szó "intelligencia" és a szó "ember".

**Example:**
A mesterséges intelligencia
Intelligencia, melyet ember alkot
Számítógépes törvényszerűségekkel
És emberi szándékokkal

**Your poem:**
(Kérjük, írd be a versedet a következő formátumban


In [ ]:
response = client.text_generation(
    f"A következő szöveg egy számla OCR-es feldolgozása után keletkezett. Készíts json kimenetet a következő szövegről: {szoveg}",
    max_new_tokens=500,
    temperature=0.1
)

print(response)



A szöveg feldolgozása után a következő json kimenetet kell létrehozni:
```json
{
  "invoice": {
    "issuer": "Adi Magyarország Élelmiszer Bt.",
    "address": "2051 Biatorbágy, Mészárosok útja 2.",
    "vat_number": "22934663-3",
    "date": "1799-01-52",
    "items": [
      {
        "description": "B00 ME Ek",
        "quantity": 5,
        "price": 375
      },
      {
        "description": "Ananász",
        "quantity": 9809,
        "price": 999
      },
      {
        "description": "Almapüré",
        "quantity": 99,
        "price": 299
      },
      {
        "description": "Zabpehely",
        "quantity": 9009,
        "price": 399
      },
      {
        "description": "Ketchup",
        "quantity": 9999,
        "price": 790
      },
      {
        "description": "Brokkoli",
        "quantity": 4509,
        "price": 649
      },
      {
        "description": "Camembert",
        "quantity": 2909,
        "price": 4180
      },
      {
        "description": "Tea"

In [ ]:
tabulat

{'level': [1,
  2,
  3,
  4,
  5,
  4,
  5,
  4,
  5,
  4,
  5,
  4,
  5,
  5,
  4,
  5,
  4,
  5,
  2,
  3,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  5,
  2,
  3,
  4,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  2,
  3,
  4,
  5,
  5,
  5,
  2,
  3,
  4,
  5,
  5,
  5,
  5,
  5,
  2,
  3,
  4,
  5,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  4,
  5,
  4,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
  5,
  5,
  5,
  4,
  5,
  5,
 

##Professzionális megoldás a problémára: Docling

In [ ]:
print(pytesseract.image_to_data(kep, lang="hun", output_type=pytesseract.Output.DICT))

{'level': [1, 2, 3, 4, 5, 4, 5, 4, 5, 4, 5, 4, 5, 5, 4, 5, 4, 5, 2, 3, 4, 5, 5, 5, 5, 4, 5, 5, 4, 5, 5, 5, 2, 3, 4, 5, 5, 5, 4, 5, 5, 4, 5, 5, 4, 5, 5, 5, 5, 2, 3, 4, 5, 5, 5, 2, 3, 4, 5, 5, 5, 5, 5, 2, 3, 4, 5, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 4, 5, 5, 4, 5, 4, 5, 4, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 4, 5, 5, 5, 4, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 5, 5, 5, 4, 5, 5, 4, 5, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 5, 4, 5, 5, 5, 5, 5, 4, 5, 5, 4, 5, 5, 5, 5, 4, 5, 4, 5, 5, 4, 5, 4, 5, 4, 5, 5, 2, 3, 4, 5, 2, 3, 4, 5, 5, 5, 4, 5, 5, 5], 'page_num': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 

Prompt: A következő számla szövegben keresd meg szolgáltató adatait és a fizetendő összeget készíts belőle JSON objektumot:
szöveg:

  
    
  


Adi Magyarország Élelmiszer Bt.
2051 Biatorbágy
Mészárosok útja 2.

102. sz, üzlet
7634 ker
Nyugati Ipari
ADÓSZÁM: 22934663- 3. 44

— WIN —

B00 ME Ek 1799 1.52

5 DB x 75 Ft/DB 375
(00 sza nö.olaj . 1299 399
Ü S; Hi 999
BŰN Félzs. túró 2509 449
500 Félzs. túró 2509 449
B00 Tejföl 207 4509 469
T 499
619
619
49 Ft/DB 1347
00. Ananász 9809 999
COD Ananász 9809 999
C00 Almapüré a 99
9 299
SZ 9 399
21 699
CO P 4 429
C00 Ananász 5809 944
C00 Zabpehely 9009 399
(00 ketchup 9999 Üz
Bo ki ő . ex. Sz, 790m! 2799
OVáSZOS CIPÓ
3 DB x 499 Ft/DB 1497
C00 Brokkoli 4509 649
BD Fr .camemb. 2909 4 180
an 8 tea 1do 299
urgonya Pp.
. 4107 1 Tára
728
1 459
1798
7699
21.269 Ft



21 263 Ft
21 263 Ft


Kimenet:
{
  "szolgaltato": {
    "nev": "Adi Magyarország Élelmiszer Bt.",
    "cim": "2051 Biatorbágy, Mészárosok útja 2., 102. sz, üzlet, 7634 ker",
    "adoszam": "22934663-3-44"
  },
  "fizetendo_osszeg": 21263
}


In [ ]:
{
  "szolgaltato": {
    "nev": "Adi Magyarország Élelmiszer Bt.",
    "cim": "2051 Biatorbágy, Mészárosok útja 2., 102. sz, üzlet, 7634 ker",
    "adoszam": "22934663-3-44"
  },
  "fizetendo_osszeg": 21263
}


{'szolgaltato': {'nev': 'Adi Magyarország Élelmiszer Bt.',
  'cim': '2051 Biatorbágy, Mészárosok útja 2., 102. sz, üzlet, 7634 ker',
  'adoszam': '22934663-3-44'},
 'fizetendo_osszeg': 21263}

#Feladat:

![img](https://www.infocentrum.hu/wp-content/uploads/help_images/elolegszamla_kep.png)

Elad

In [ ]:
import requests
from io import BytesIO

url = "https://www.infocentrum.hu/wp-content/uploads/help_images/elolegszamla_kep.png"

response = requests.get(url)
image_bytes = response.content

kep = Image.open(BytesIO(image_bytes))

In [ ]:
szoveg = pytesseract.image_to_string(kep,lang="hun")

# Táblázat kinyerése
tabulat = pytesseract.image_to_data(kep, lang="hun", output_type=pytesseract.Output.DICT)

# Kiíratás
#print("Szöveg:")
#print(szoveg)

print("Táblázat:")
for sor in tabulat['text']:
    print(sor)

Táblázat:




 



infocentrum

szoftver
stúdió



Előleg
számla



Sorszám:
000007/2015



 



 



Szállító

Infocentrum
Szoftver
Stúdió

7400
Kaposvár,
Vár
u.
12-14.



Adószám:
12345678-1-12

Közösségi
adószám
HU12345678

Bankszámlaszám:

tel:06-30-916-8888


fax
821312-391

wwrwinfocentrumhu



Vevő

TesztKereskedelmi
Kft.



Adószám:
12121212-1-21

Közösségi
adószám
HU12121212

Bankszámlaszám:



 



1023
Budapest
II.
kerület,
Madár
u.
55.



 



 



 



 



 



 



 



 



 



 



 



 



 



 



 



 



 



 



 



Megrendelés
száma:
Fizetés
módja:
Telj.
időpontja:
Számla
kelte:
Fiz.
határidő;

Átutalás
2015.06.12.
2015.06.12.
2015.06.12.

Egyéb
adatok:

orolási
szám,
Megnevezés
e.
ÁT]
Átaérté

VTSZ/
Besorolási
szám,
Megi
ME]
Menny.
[
ágmélken
té]
Áraérték

ELŐLEG
ri
a]
50
000
50000]
2796
13500
63
500]

4
lioenszes
hálózatos
számlázó
praram
előlege.

Ji
dési
ök
tsdlé
Számlaérték:
50.000
Ft

Áfa:
13500
Ft

27
50000
Ft
13500Ft
63500Ft

Fizetendő:
$3500Ft



 

In [ ]:
!pip install docling

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 162.6/162.6 kB 5.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48.2/48.2 kB 3.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 159.6/159.6 kB 13.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.9/115.9 kB 9.9 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.9/80.9 kB 6.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 15.1/15.1 MB 86.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.9/2.9 MB 78.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.1/42.1 kB 3.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.9/2.9 MB 81.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 244.3/244.3 kB 17.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 472.8/472.8 kB 31.3 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 541.1/541.1 kB 34.7 MB/s eta 0:00:00
   ━

In [ ]:
from pathlib import Path
from docling.backend.docling_parse_backend import DoclingParseDocumentBackend
from docling.datamodel.base_models import InputFormat
from docling.datamodel.pipeline_options import (
    EasyOcrOptions,
    OcrMacOptions,
    PdfPipelineOptions,
    RapidOcrOptions,
    TesseractCliOcrOptions,
    TesseractOcrOptions,
)
from docling.document_converter import DocumentConverter, PdfFormatOption

In [ ]:
def main():
    input_doc = Path("szamla1.jpg")

    pipeline_options = PdfPipelineOptions()
    pipeline_options.do_ocr = True
    pipeline_options.do_table_structure = True
    pipeline_options.table_structure_options.do_cell_matching = True

    # Any of the OCR options can be used:EasyOcrOptions, TesseractOcrOptions, TesseractCliOcrOptions, OcrMacOptions(Mac only), RapidOcrOptions
    # ocr_options = EasyOcrOptions(force_full_page_ocr=True)
    # ocr_options = TesseractOcrOptions(force_full_page_ocr=True)
    # ocr_options = OcrMacOptions(force_full_page_ocr=True)
    ocr_options = RapidOcrOptions(force_full_page_ocr=True)
    #ocr_options = TesseractCliOcrOptions(force_full_page_ocr=True)
    pipeline_options.ocr_options = ocr_options

    converter = DocumentConverter(
        format_options={
            InputFormat.PDF: PdfFormatOption(
                pipeline_options=pipeline_options,
            )
        }
    )

    doc = converter.convert(input_doc).document
    md = doc.export_to_markdown()
    print(md)

if __name__ == "__main__":
    main()

Szamla sorszsma: 303301754

Szanla kele:

## SLÁMLARÉSZLETEZÖ

|                 | (Ft)   |    |          |          |
|-----------------|--------|----|----------|----------|
|                 |        | 27 | 425,86   |          |
| [ulladelkerele; |        | 27 |          |          |
|                 |        | 27 | 425,88   |          |
|                 |        |    |          | Qlo      |
|                 |        |    | 1 278,00 |          |
|                 |        |    |          | 6 010,00 |

1 .


In [ ]:
import os
from huggingface_hub import snapshot_download
from docling.datamodel.pipeline_options import PdfPipelineOptions, RapidOcrOptions
from docling.document_converter import (
    ConversionResult,
    DocumentConverter,
    InputFormat,
    PdfFormatOption,
)



def main():
    # Source document to convert
    source = "szamla.jpg"

    # Download RappidOCR models from HuggingFace
    print("Downloading RapidOCR models")
    download_path = snapshot_download(repo_id="SWHL/RapidOCR")

    # Setup RapidOcrOptions for english detection
    det_model_path = os.path.join(
        download_path, "PP-OCRv4", "en_PP-OCRv3_det_infer.onnx"
    )
    rec_model_path = os.path.join(
        download_path, "PP-OCRv4", "ch_PP-OCRv4_rec_server_infer.onnx"
    )
    cls_model_path = os.path.join(
        download_path, "PP-OCRv3", "ch_ppocr_mobile_v2.0_cls_train.onnx"
    )
    ocr_options = RapidOcrOptions(
        det_model_path=det_model_path,
        rec_model_path=rec_model_path,
        cls_model_path=cls_model_path,
    )

    pipeline_options = PdfPipelineOptions(
        ocr_options=ocr_options,
    )

    # Convert the document
    converter = DocumentConverter(
        format_options={
            InputFormat.PDF: PdfFormatOption(
                pipeline_options=pipeline_options,
            ),
        },
    )

    conversion_result: ConversionResult = converter.convert(source=source)
    doc = conversion_result.document
    md = doc.export_to_markdown()
    print(md)


if __name__ == "__main__":
    main()

## TOM GREEN HANDYMAN

5 Street, That Area Code Telephone: 0800 XXX XXX Any Any City,

Date :

6/5/2016

Invoice No

0003521

Tax Registered No

123456

Mr and Mrs Fielding This Address This This Area Code City

## TAX INVOICE

| Quantity   | Description              | Unit Price   | Cost    |
|------------|--------------------------|--------------|---------|
|            | Upgrade to Bathroom      |              |         |
| 23.75      | Labour                   | 40.00        | 950.00  |
| 50         | Nails and screws         | 0.80         | 40.00   |
| 1          | Paint and Plywood        | 1000.00      | 1000 00 |
| 40         | Imported wall tiles      | 14 00        | 560.00  |
| 1          | Freight                  | 150.00       | 150 00  |
|            | Sub-contractor : Tile-lt |              | 228.00  |
|            |                          | Subtotal     | 2928.00 |
|            |                          | Tax          | 439.20  |
|            |                     

In [ ]:
import yfinance as yf
import pandas as pd
# Define the ticker symbols
tickers = ['AAPL', 'GOOG', 'MSFT']
# Define the start and end dates
start_date = '2020-01-01'
end_date = '2022-12-31'
# Create an empty DataFrame to store the data
data = pd.DataFrame()
# Download the data
for ticker in tickers:
 df = yf.download(ticker, start=start_date, end=end_date, interval='1mo')
 df['Ticker'] = ticker # Add a column with the ticker symbol
 data = pd.concat([data, df])
# Reset the index
data.reset_index(inplace=True)
# Save the data to a CSV file
data.to_csv('stock_data.csv', index=False)

YF.download() has changed argument auto_adjust default to True


[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


In [ ]:
!pip install openai

In [ ]:
from google.colab import userdata
api_key = userdata.get('openai_api')

In [ ]:
import os
from openai import OpenAI
client = OpenAI(api_key=api_key)

import pandas as pd

csv_string = data[:2].to_string(index=False)


input_text = f"adok egy sor céginformációt a következő sorrendben Cég,Ár-arányos nyereség,Ár-arányos könyv szerinti érték,Saját tőke megtérülése %,Adósság a saját tőkéhez,Aktuális mutató,Bruttó árrés %,elemezze az egyes cégek mutatóit, hivatkozzon a cég nevére, és írjon egy tömör választ magyarul.\n\n{csv_string}"

# Use client.chat.completions.create for chat models
response = client.chat.completions.create(
    model="gpt-4o",  # or "gpt-3.5-turbo" for a faster, cheaper model
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": input_text},
    ],
    max_tokens=400  # Limiting the length of the generated text
)

generated_text = response.choices[0].message.content.strip()
print("GPT-4o Cégelemzés:", generated_text)

GPT-4o Cégelemzés: A következő céginformációk értékelése a megadott mutatók alapján segíthet megérteni, hogyan teljesít egy vállalat és milyen befektetési lehetőséget kínál.

1. **AAPL (Apple Inc.)**:
   - **Ár-arányos nyereség (P/E ratio)**: Ha alacsony, az azt jelentheti, hogy a részvény alulértékelt, vagy hogy a vállalat enyhén növekszik. Ha magas, az utalhat a jövőbeli növekedési várakozásokra.
   - **Ár-arányos könyv szerinti érték (P/B ratio)**: Az Apple esetén, ha alacsony, a részvény alulértékelt lehet a vagyonához képest. Ha magas, az sugallhatja, hogy a piac optimistán ítéli meg a cég eszközeit.
   - **Saját tőke megtérülése (ROE)**: Ha magas, az azt jelzi, hogy hatékonyan használja fel a tőkét a profit termelésére.
   - **Adósság a saját tőkéhez (Debt to Equity)**: Alacsony mutató jó pénzügyi stabilitást, míg magas mutató potenciális kockázatot jelenthet.
   - **Aktuális mutató (Current Ratio)**: 1 felett azt mutatja, hogy a cég képes fedezni rövid távú kötelezettségeit.
   